name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/ComdirectPDFExtractor.java

```diff
@@ -1809,249 +1822,177 @@ public List<Item> postProcessing(List<Item> items)
-                }
-                else if (transactions.size() == 2)
-                {
-                    var dividendTransaction = (AccountTransaction) transactions.get(0).getSubject();
-                    var taxesTransaction = (AccountTransaction) transactions.get(1).getSubject();
-
-                    // Memory the Item with taxes properties
-                    Item taxesTransactionItem = transactions.get(1);
-
-                    // Which transaction is the taxes and which the dividend?
-                    if (!AccountTransaction.Type.TAXES.equals(taxesTransaction.getType()))
-                    {
-                        dividendTransaction = (AccountTransaction) transactions.get(1).getSubject();
-                        taxesTransaction = (AccountTransaction) transactions.get(0).getSubject();
-                        taxesTransactionItem = transactions.get(0);
-                    }
-
-                    // Check if there is a dividend transaction and a taxes transaction
-                    if (AccountTransaction.Type.TAXES.equals(taxesTransaction.getType())
-                                    && AccountTransaction.Type.DIVIDENDS.equals(dividendTransaction.getType()))
-                    {
-                        // @formatter:off
-                        // Set the correct gross amount if the taxes base (gross)
-                        // is greater than the calculated gross amount (withholding tax).
-                        // @formatter:on
-                        Money grossTaxesTreatment = (Money) taxesTransactionItem.getData(ATTRIBUTE_GROSS_TAXES_TREATMENT);
+                saleTransaction.getPortfolioTransaction().addUnit(new Unit(Unit.Type.TAX, taxesTransaction.getMonetaryAmount()));
 
-                        if (grossTaxesTreatment != null
-                                        && grossTaxesTreatment.isGreaterThan(dividendTransaction.getGrossValue()))
-                        {
-                            dividendTransaction.setMonetaryAmount(grossTaxesTreatment);
+                if (!saleTransaction.getSource().equals(taxesTransaction.getSource()))
+                    saleTransaction.setSource(saleTransaction.getSource() + "; " + taxesTransaction.getSource());
 
-                            ExtractorUtils.fixGrossValue().accept(dividendTransaction);
-                        }
+                saleTransaction.setNote(concat(saleTransaction.getNote(), taxesTransaction.getNote()));
 
-                        // Subtract the taxes from the taxes transaction from the total amount
-                        dividendTransaction.setMonetaryAmount(dividendTransaction.getMonetaryAmount()
-                                        .subtract(taxesTransaction.getMonetaryAmount()));
-
-                        // Add taxes as tax unit
-                        dividendTransaction.addUnit(new Unit(Unit.Type.TAX, taxesTransaction.getMonetaryAmount()));
+                items.removeIf(item -> item == pair.tax());
```

Comment:
Isn't that just removing one item from the list?

So just:
```
items.remove(pair.tax());
```

RemoveIf iterates over the full list to check the condition which will be true only once - for the tax item